pkg/templates/kubeadm/v1beta3/kubeadm.go

```diff
@@ -48,20 +48,25 @@ const (
 )
 
 const (
-	// fixedEtcdVersion is an etcd version that doesn't have known data integrity and durability bugs
+	// minimumFixedEtcdVersion is an etcd version that doesn't have known data integrity and durability bugs
 	// (see etcdVersionCorruptCheckExtraArgs for more details)
-	fixedEtcdVersion = "3.5.6-0"
+	minimumFixedEtcdVersion = "3.5.8-0" // at least this version has to be installed
+	installFixedEtcdVersion = "3.5.9-0" // if minimum fixed version is not detected, install this version
 
-	// NB: Currently no Kubernetes version uses 3.5.6, but to avoid deleting the code
+	// NB: Currently no Kubernetes version uses 3.5.8, but to avoid deleting the code
 	// we just use some super high version as a fixed version.
 	// fixedEtcd123 defines a semver constraint used to check if Kubernetes 1.23 uses fixed etcd version
 	fixedEtcd123 = ">= 1.23.15, < 1.24"
 	// fixedEtcd124 defines a semver constraint used to check if Kubernetes 1.24 uses fixed etcd version
 	fixedEtcd124 = ">= 1.24.9, < 1.25"
 	// fixedEtcd125 defines a semver constraint used to check if Kubernetes 1.25 uses fixed etcd version
 	fixedEtcd125 = ">= 1.25.5, < 1.26"
-	// fixedEtcd126 defines a semver constraint used to check if Kubernetes 1.26+ uses fixed etcd version
-	fixedEtcd126 = ">= 1.26.0"
+	// fixedEtcd126 defines a semver constraint used to check if Kubernetes 1.26 uses fixed etcd version
+	fixedEtcd126 = ">= 1.26.0, < 1.27"
+	// fixedEtcd127 defines a semver constraint used to check if Kubernetes 1.27 uses fixed etcd version
+	fixedEtcd127 = ">= 1.27.0, < 1.28"
+	// fixedEtcd128 defines a semver constraint used to check if Kubernetes 1.28+ uses fixed etcd version
+	fixedEtcd128 = ">= 1.28.0"
```

Comment:
Let's for now use `>= .0` (e.g. `>= 1.24.0`) everywhere because no Kubernetes patch release currently uses `3.5.9-0` (that's hopefully going to change soon).

Reply:
`3.5.9-0` is has been backported to 1.25/26/27, I adapted the code and tests accordingly.